src/make-http-request.coffee

```diff
@@ -43,9 +43,7 @@ makeHTTPRequest = (method, url, data, headers = {}, modify) ->
       if error?.status is 408
         resolve makeHTTPRequest.apply null, originalArguments
       else if error?
-        # Prefer rejecting with the response, since it'll have more specific information.
-        # TODO: Reject with the error as expected and access the response through `error.response` in the handler.
-        reject response ? error
+        reject error
```

Comment:
this highlights another good question. currently the panoptes-javascript-client (p-j-c) just returns an error message, but with this PR and the related p-j-c PR it'll return an error object that includes status and response attributes. the response attribute includes the verbose error message currently returned.

recently I think there was an error related to If-Then headers and the API I didn't quite understand, but the point was made that maybe our error messages should be less verbose or more generic?

if this PR and the related p-j-c PR are merged as is, the error message the PFE will throw to the console will be a less verbose version, but the more verbose explanation will be available if we want to access it.

the example I've been playing with is requesting a project by ID that doesn't exist. here's the `error.response` attribute:
![screen shot 2017-03-14 at 3 25 40 pm](https://cloud.githubusercontent.com/assets/12118751/24155917/1347d09c-0e23-11e7-93d1-a9ae5b97a072.png)
currently the "Could not find project with id-'1693777'" message is returned, but with these changes just "404: Not Found" would be console logged, though the more verbose one would be available as shown above.

If we want to go the route of returning the verbose error message just as we have been, I can clean [this p-j-c PR](https://github.com/zooniverse/panoptes-javascript-client/pull/61) up that I closed?